runtime/lua/vim/_editor.lua

```diff
@@ -307,14 +311,26 @@ end
 ---                            If a table, executes a single command. In this case, it is an alias
 ---                            to |nvim_cmd()| where `opts` is empty.
 ---@see |ex-cmd-index|
-function vim.cmd(command)
-  if type(command) == 'table' then
-    return vim.api.nvim_cmd(command, {})
-  else
-    return vim.api.nvim_exec(command, false)
-  end
+function vim.cmd(command) -- luacheck: no unused
+  error(command) -- Stub for gen_vimdoc.py
 end
 
+vim.cmd = setmetatable({}, {
+  __call = function(_, command)
+    if type(command) == 'table' then
+      return vim.api.nvim_cmd(command, {})
+    else
+      return vim.api.nvim_exec(command, false)
+    end
+  end,
+  __index = function(t, command)
+    t[command] = function(...)
+      return vim.api.nvim_cmd({ cmd = command, args = { ... } }, {})
```

Comment:
>  `vim.cmd.write("myfile.txt", { force = true })`

This one is awkward, but we can do `vim.cmd.write{ "myfile.txt", force = true }`

> `vim.cmd.substitue("/hi/hello/g", { range = { "%"} })`

I'm not sure about this one. I think you are better off with `vim.cmd{cmd="s/hi/hello/g", range = { "%"} }`

Reply:
> `vim.cmd.write { force = true, args = { "myfile.txt" } }`

`force` will be spelled `bang` I assume? Let's use the same names as the API unless there's a very good reason not to.

Reply:
> I think the interface should be good enough so the user doesn't have to fallback to vim.cmd unless they are trying to do multiple commands at the same time (something this new design is not for). Otherwise we get overhead, the user trying to decide if using the "indexed" version is going to work for his use case.

Actually, `substitue` allows whitespace so `vim.cmd.substitute('/hi/hello/g')` already works.